File: vian.py
a = "37"
p1 = 0
p2 = 1

while "077201" not in a[len(a) - 10:]:
  if len(a) % 100000 == 0:
    print(len(a), p1, p2)
  a_p1 = int(a[p1])
  a_p2 = int(a[p2])
  next_a = a_p1 + a_p2
  a += str(next_a)
  p1 = (p1 + a_p1 + 1) % len(a)
  p2 = (p2 + a_p2 + 1) % len(a)
print(a[5:15])
print(a[18:28])
print(a[2018:2028])
print(a[330121:330131])
print(a.index("077201"))

File: vian.rb
a = "37"
p1 = 0
p2 = 1

until a.size >= 10 || (a[(a.size - 10)..a.size].include?  "077201")
  a_p1 = a[p1].to_i
  a_p2 = a[p2].to_i
  next_a = a_p1 + a_p2
  a += next_a.to_s
  p1 = (p1 + a_p1 + 1) % a.size
  p2 = (p2 + a_p2 + 1) % a.size
end
puts(a.index("077201"))
